privacyscore/backend/management/commands/scanfromfile.py

```diff
@@ -28,22 +28,33 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('file_path')
         parser.add_argument('-s', '--sleep-between-scans', type=float, default=0)
+        parser.add_argument('-c', '--create-list-name')
 
     def handle(self, *args, **options):
         if not os.path.isfile(options['file_path']):
             raise ValueError('file does not exist!')
-        self.stdout.write('Reading from file {}'.format(options['file_path']))
+        scan_list = None
+
+        if options['create_list_name']:
```

Comment:
Spontaneously, I don't like branching here.
If you branch in the end then you don't get only half the list if, say, something failed when scanning the sites. Also, you probably get more compact code by doing smth like `ScanList.objects.create(sites=sites, is_private=True)` or so.

Reply:
If I branch in the end, and get to know that the ScanList already exists, we would have time wasted for reading the whole file for e.g. with 100 000 entries. We obviously want a list created when we are using the  `create_list_name` parameter.
In the end the main requirement is to have all the entries from the file in the list.  If something failed while scanning a website, it might be successful in the `rescan` phase.

Reply:
> We obviously want a list created when we are using the create_list_name parameter.

exactly. That's why I would expect there to be a "ScanList.objects.create"  call. We don't force names to be unique and I think it's a very fine thing.



> If something failed while scanning a website, it might be successful in the rescan phase.

But I don't see how the list is edited during a rescan phase. Conceptually, I believe a "rescan" shouldn't involve modifying an existing list. I expect the rescan to only trigger the list's `scan` function. So you don't get to add new sites to the list.

Reply:
I will probably branch after reading all the sites from the file because I think it's important to  have all the sites in the ScanList before we start to scan.
I will change the List name to be not unique. But the  `ScanList.objects.create(sites=sites, is_private=True)` cannot still be done in 1 line because of many-to-many relationship.

Reply:
I will probably branch after reading all the sites from the file because I think it's important to  have all the sites in the ScanList before we start to scan.
I will change the List name to be not unique. But the  `ScanList.objects.create(sites=sites, is_private=True)` cannot still be done in 1 line because of many-to-many relationship.